src/main/java/net/snowflake/ingest/connection/SecurityManager.java

```diff
@@ -110,14 +120,26 @@
    * regenerateToken - Regenerates our Token given our current user,
    *                    account and keypair
    */
-  private void regenerateToken()
+  public void regenerateToken()
   {
     //create our JWT claim object
     JwtClaims claims = new JwtClaims();
 
     //set the issuer to the fully qualified username
-    claims.setIssuer(account + "." + user);
-    LOGGER.info("Creating Token with Issuer {}", account + "." + user);
+    String publicKeyFPInJwt = null;
+    try
+    {
+      publicKeyFPInJwt = getPublicKeyFp(savePublicKey(keyPair.getPublic()));
+      publicKeyFingerPrint = publicKeyFPInJwt;
+    }
+    catch(Exception e)
+    {
+      LOGGER.info("Failed to Generate public key fingerprint");
```

Comment:
Just logging info level probably not enough. Generating public key fingerprint is pretty important, without public key fingerprint, the whole key pair authentication will fail. So just catching the exception and let it go through is probably not a good idea. Maybe throw some exception is better. Adding a class called SnowflakeException and indicating a `INTERNALL_ERROR`